assets/js/components/ModulesList.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * ModulesList component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import map from 'lodash/map';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	showErrorNotification,
+	trackEvent,
+} from '../util';
+import Data from 'googlesitekit-data';
+import Link from './Link';
+import ModuleIcon from './ModuleIcon';
+import GenericError from './notifications/generic-error';
+import ModuleSettingsWarning from './notifications/module-settings-warning';
+import VisuallyHidden from './VisuallyHidden';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ModulesList( { moduleSlugs } ) {
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const modulesData = useSelect( ( select ) => select( CORE_MODULES ).getModules() ?? {} );
+
+	/**
+	 * Handle setup module click event.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param {string} slug Module slug.
+	 */
+	const handleSetupModule = useCallback( async ( slug ) => {
+		try {
+			const { response } = await activateModule( slug );
```

Comment:
The `try/catch` here is no longer working as it was before as neither `activateModule` or `trackEvent` will throw. Instead, `error` should come from the return value of `activateModule`, just as it does with our other async actions.